dgraph/cmd/zero/raft.go

```diff
@@ -88,11 +88,14 @@ func (n *node) AmLeader() bool {
 // {2 bytes Node ID} {4 bytes for random} {2 bytes zero}
 func (n *node) initProposalKey(id uint64) error {
 	x.AssertTrue(id != 0)
-	b := make([]byte, 8)
-	if _, err := rand.Read(b); err != nil {
+	random4Bytes := make([]byte, 4)
+	if _, err := rand.Read(random4Bytes); err != nil {
 		return err
 	}
-	proposalKey = n.Id<<48 | binary.BigEndian.Uint64(b)<<16
+	proposalKey = n.Id<<48 | uint64(binary.BigEndian.Uint32(random4Bytes))<<16
+	// We want to avoid spillage to node id in case of overflow. So by setting 48th bit to
```

Comment:
Maybe add more details of why there could be spillage and why setting 48th bit to 0 would solve it. It may not be immediately obvious to the person reading the code.